### PR TITLE
Update @appland/navie to include code editor

### PR DESCRIPTION
**Pull request by ahtrotta**
This is part of [this PR](https://github.com/getappmap/appmap-js/pull/1706), but I want to release a new version of `@appland/navie` first, so that `@appland/cli` can bump it's dependencies using the new `@appland/navie` version.

### CODE REVIEW

**Review by dustinbyrne — CHANGES_REQUESTED** [review]
I think it's worth sorting out the duplicate array. Also, it'd be worth adding a test.

File: packages/navie/src/services/project-info-service.ts [hunk 1]
```diff
@@ -33,6 +33,12 @@ export default class ProjectInfoService {
         () => projectInfo.some(({ appmapStats }) => appmapStats.numAppMaps > 0),
         `The project does not contain any AppMaps.`,
       ],
+      [
+        PromptType.CodeEditor,
+        PromptType.CodeEditor,
+        () => projectInfo.some(({ codeEditor }) => Boolean(codeEditor)),
+        `The project does not contain any code editor information.`,
+      ],
```
[the remark this answers is not in the document]
Comment (kgilpin) [reply packages/navie/src/services/project-info-service.ts:41]: Yeah, if it's not there just ignore this section. The user may be using an outdated client, or somehow calling Navie from a different environment (custom plugin for neovim? :-))

Unlike with missing project info, it's not something we need to try and direct the AI to take action on.

File: packages/navie/src/services/project-info-service.ts [hunk 2]
```diff
@@ -45,7 +51,7 @@ export default class ProjectInfoService {
       const promptValue = projectInfo.map((info) => {
         let value: Record<string, unknown> | undefined = info[projectInfoKey];
         const hasName = 'name' in info;
-        if (!hasName && info.appmapConfig?.name) {
+        if (!hasName && info.appmapConfig?.name && promptType !== PromptType.CodeEditor) {
           value = { ...value, name: info.appmapConfig.name };
         }
         return value;
```
[the remark this answers is not in the document]
Comment (kgilpin) [reply packages/navie/src/services/project-info-service.ts:57]: Also we want that to be like `codeEditorName` rather than just `name`

**Comment by kgilpin** [comment]
> I think it's worth sorting out the duplicate array. Also, it'd be worth adding a test.

Tests of the prompt behavior would be really useful, and handy to place in the PR description because anything that affects the prompt is super important.

**Comment by ahtrotta** [comment]
No longer relevant

**Pull request closed without merging**